### DISCUSSION TITLE
GPU using from cli

### DISCUSSION THREAD

**Discussion by jooni22** [absadiki/subsai#9] [General]
Is it possible to use subsai from console (bash) to use GPU instead of CPU?

**Comment by absadiki** [comment]
Yes it is possible if the model supports GPU, 
Actually you can use any parameter in the model-configs argument 
For `openai/whisper` for example, you can check the list of all parameters [here](https://abdeladim-s.github.io/subsai/#subsai.models.whisper_model.WhisperModel.config_schema) 
and you can use it in from the CLI as follows : 
```shell
subsai ./assets/test1.mp4 --model openai/whisper --model-configs '{"model_type": "small", "device": "cuda"}' --format srt
```

Hope this helps ?

**Comment by jooni22** [comment]
Yep, that working as well. I had issue with Nvidia and after fixing the subsai automatically set cuda if model supports GPU. Many thanks

**Reply by absadiki** [reply]
you're welcome :)

**Comment by gandarufu** [comment]
Which version of Cuda and cuDNN do you suggest? I currently tried 12.1 and i can't get subsAI to run on my gpu.

**Reply by s-h-a-d-o-w** [reply]
It's been 2 years but... this is still a problem and in case somebody else stumbles across it, this is how I made it work:

1. Wipe all globally installed packages. (Don't know whether that's relevant if you're using a virtual environment, I rarely work with Python.)
2. Optional: Create a virtual environment
```bash
python -m venv subsai
source subsai/bin/activate  # On Windows: subsai\Scripts\activate
```
3. Install subsai
4. Check CUDA version: `nvcc --version`
5. Get the appropriate installer command [from the PyTorch website](https://pytorch.org/get-started/locally/) and run it.
6. Run subsai.

(I then got numerous warnings because different things rely on different, incompatible dependency versions but... it works.)

**Reply by absadiki** [reply]
Thanks @s-h-a-d-o-w
